providers/class.two-factor-email.php

```diff
@@ -167,6 +188,22 @@ public function authentication_page( $user ) {
 	}
 
 	/**
+	 * Send the email code, if missing or requested.
+	 *
+	 * @param  WP_USer $user WP_User object of the logged-in user.
+	 *
+	 * @return boolean       If the two factor should be authenticated.
+	 */
+	public function process_request( $user ) {
+		if ( isset( $user->ID ) && isset( $_POST['two-factor-email-code-resend'] ) ) {
```

Comment:
Note that the 2FA request nonce has been already verified before this is called.